### ISSUE TITLE
Have you tried to generate rectangular images?

### ISSUE THREAD

**Issue by wdf19961118** [NVlabs/stylegan2-ada-pytorch#94]
The images of my dataset are not square, if I crop the image then some information will be lost. Also, I can resize the image without cropping, but the shape of objects will be changed. So, is changing the network is the only way to solve the problem?

**Comment by PDillis** [comment]
Yes, the classical solution is to modify the code in order to solve this. Another option is you can use the `--transform=center-crop-wide` option when running `dataset_tool.py`. This will make a rectangular center-crop of your image and then pad with black on top and bottom of the cropped image, resulting in a square one. Try it out in a couple of images or something before applying it to your whole dataset, as the result might not be what you want (mainly due to the center-crop part which will change some proportions).

**Comment by wdf19961118** [comment]
> Yes, the classical solution is to modify the code in order to solve this. Another option is you can use the `--transform=center-crop-wide` option when running `dataset_tool.py`. This will make a rectangular center-crop of your image and then pad with black on top and bottom of the cropped image, resulting in a square one. Try it out in a couple of images or something before applying it to your whole dataset, as the result might not be what you want (mainly due to the center-crop part which will change some proportions).

Thanks for your advice. Cause that cropping operations leads to information loss , and have you try to directly pad the rectangular images without any cropping operations?

**Comment by PDillis** [comment]
You can define your own transformation in `dataset_tool.py`, specifically in the `make_transform` function. I assume you would want to first pad with black to make the image a square, then resize it to a specific size (`1024`, `512`, etc.). If I had the code I would share it, but it's not yet a priority and I've only implemented a `center-crop-tall` in case your images should be padded with black on the sides and not top and bottom (you can find the code [here](https://github.com/PDillis/stylegan2-ada-pytorch/blob/dab4f40a4ec1031b55f2c4b8e59e530bd47e2cd1/dataset_tool.py#L219), use it as an example on how to implement your `resize-pad` transform, or however you want to call it).

In any case, you should first try with a subset of your data (1 image even) to see if that's what you want before implementing it in your full dataset. I'd also suggest you try the `center-crop-wide` in case you don't lose that much information,

**Comment by GilesBathgate** [comment]
See also #52
